lib/cisco_node_utils/dns_domain.rb

```diff
@@ -0,0 +1,64 @@
+#
+# NXAPI implementation of DnsDomain class
+#
+# September 2015, Hunter Haugen
+#
+# Copyright (c) 2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# "group" is a standard SNMP term but in NXOS "role" is used to serve the
+# purpose of group; thus this provider utility does not create snmp groups
+# and is limited to reporting group (role) existence only.
+
+require File.join(File.dirname(__FILE__), 'node_util')
+
+module Cisco
+  # DnsDomain - node utility class for DNS search domain config management
+  class DnsDomain < NodeUtil
+    attr_reader :name
+
+    def initialize(name, instantiate=true)
+      unless name.is_a? String
+        fail TypeError, "Expected a string, got a #{name.inspect}"
```

Comment:
Inspect will not print helpful information for some classes.  I recommend using class.name instead.

```
irb
1.9.3-p551 :001 > test = 123
 => 123 
1.9.3-p551 :002 > test.inspect
 => "123" 
1.9.3-p551 :003 > test.class.name
 => "Fixnum"
```